### PR TITLE
[SAC-207] Support for S3 objects

### PR DESCRIPTION
**Pull request by WalkerTR**
## What changes were proposed in this pull request?

**Support for AWS S3 paths**
Paths with scheme `s3`, `s3a` and `s3n` are converted to appropriate atlas entities.

```scala
scala> Seq(1,2).toDF().write.save("s3a://dongjoon-public/s3")
```
screen

## How was this patch tested?

Manual testing and unit tests.

### CODE REVIEW

**Comment by kr7ysztof** [comment]
it would be great to have it.

**Comment by Bongani** [comment]
@dongjoon-hyun & @HeartSaVioR could you please review this pull request when you have a moment.

**Comment by dongjoon-hyun** [comment]
Hi, @Bongani .
Your first contribution (#204) is merged, could you do the followings?
1. Create an issue about this.
2. Update the title of this PR by using the issue ID like `[SAC-XXX]` (please refer your first commit).
3. Update the PR description with screenshot of S3.
4. Of course, you need to resolve the conflicts and should not have irrelevant content like test coverage plugin in this PR.

File: pom.xml [hunk 1]
```diff
@@ -44,6 +44,7 @@
     <mockito.version>1.10.19</mockito.version>
     <integration.test.enabled>false</integration.test.enabled>
     <jersey.version>1.19</jersey.version>
+    <cobertura.version>2.7</cobertura.version>
```
Comment (dongjoon-hyun) [inline pom.xml]: Please remove this.

**Comment by dongjoon-hyun** [comment]
Ur, could you resolve the conflicts?

**Pull request merged**